src/gui/src/dbDescriptors.cpp

```diff
@@ -527,6 +529,249 @@ bool DbNetDescriptor::getBBox(std::any object, odb::Rect& bbox) const
   return false;
 }
 
+void DbNetDescriptor::drawPathSegment(odb::dbNet* net, const odb::dbObject* sink, Painter& painter) const
```

Comment:
This feels a bit long - I would split it into 2-3 pieces.

Reply:
split into 3 functions